packages/terra-functional-testing/tests/jest/services/wdio-terra-service.test.js

```diff
@@ -156,223 +161,201 @@ describe('WDIO Terra Service', () => {
     expect(global.Terra.serviceOptions).toEqual(expectedServiceOptions);
   });
 
-  it('should upload screenshots in onComplete', () => {
-    const localConfig = {
-      serviceOptions: {
-        selector: 'mock-selector',
-        useRemoteReferenceScreenshots: true,
-        buildBranch: BUILD_BRANCH.master,
-        buildType: BUILD_TYPE.branchEventCause,
-      },
-    };
+  describe('Screenshot stuff', () => {
+    // Mocking out this method lets us test github API scenarios.
+    const request = jest.fn();
+    createOctokit.mockImplementation(() => ({
+      request,
+    }));
 
-    const runnerConfig = {
+    // Adding all the stuff we need to test remote screenshot config.
+    const rsConfig = {
+      ...config,
       screenshotsSites: {
         repositoryId: 'mock-repositoryId',
         repositoryUrl: 'mock-repositoryUrl',
       },
-      getRemoteScreenshotConfiguration: jest.fn().mockImplementation(() => {}),
+      getRemoteScreenshotConfiguration: jest.fn(() => ({ publishScreenshotConfiguration: 1 })),
     };
 
-    const service = new WdioTerraService({}, {}, localConfig);
-
-    service.onComplete({}, runnerConfig);
-
-    expect(runnerConfig.getRemoteScreenshotConfiguration).toHaveBeenCalledWith(runnerConfig.screenshotsSites, localConfig.serviceOptions.buildBranch);
-  });
-
-  it('should not upload screenshots in onComplete if buildBranch is a pullRequest', () => {
-    const localConfig = {
-      serviceOptions: {
-        selector: 'mock-selector',
-        useRemoteReferenceScreenshots: true,
-        buildBranch: 'pr-31',
-        buildType: BUILD_TYPE.branchEventCause,
-      },
-    };
-
-    const service = new WdioTerraService({}, {}, localConfig);
-
-    service.onComplete({}, config);
-
-    expect(config.getRemoteScreenshotConfiguration).not.toHaveBeenCalledWith();
-  });
-
-  it('should not upload screenshots in onComplete if buildType is not BranchEventCause', () => {
-    const localConfig = {
-      serviceOptions: {
-        selector: 'mock-selector',
-        useRemoteReferenceScreenshots: true,
-        buildBranch: BUILD_BRANCH.master,
-        buildType: undefined,
-      },
-    };
-
-    const service = new WdioTerraService({}, {}, localConfig);
-
-    service.onComplete({}, config);
-
-    expect(config.getRemoteScreenshotConfiguration).not.toHaveBeenCalledWith();
-  });
-
-  describe('github comment onComplete', () => {
-    const OLD_ENV = process.env;
-
-    beforeEach(() => {
-      jest.resetModules(); // Most important - it clears the cache
-      process.env = { ...OLD_ENV }; // Make a copy
-    });
-
-    afterAll(() => {
-      process.env = OLD_ENV; // Restore old environment
-    });
-
-    it('should post to github', async () => {
-      process.env.SCREENSHOT_MISMATCH_CHECK = true;
+    it('should upload screenshots in onComplete', async () => {
```

Comment:
Note to reviewers: I didn't remove and tests, I just grouped them and moved some consts around.

Reply:
Many more tests now.